servo/connectors/kubernetes.py

```diff
@@ -2003,6 +2016,578 @@ async def get_restart_count(self) -> int:
 
         return count
 
+# Workarounds to allow use of api_client.deserialize() public method instead of private api_client._ApiClient__deserialize
+# TODO: is this workaround worth it just to avoid using the private method?
+# fix for https://github.com/kubernetes-client/python/issues/977#issuecomment-594045477
+def default_kubernetes_json_serializer(o: Any) -> Any:
+    if isinstance(o, (datetime.datetime, datetime.date)):
+        return o.isoformat()
+    raise TypeError(f'Object of type {o.__class__.__name__} '
+                        f'is not JSON serializable')
+
+# https://github.com/kubernetes-client/python/issues/977#issuecomment-592030030
+class FakeKubeResponse:
+    """Mocks the RESTResponse object as a workaround for kubernetes python api_client deserialization"""
+    def __init__(self, obj):
+        self.data = json.dumps(obj, default=default_kubernetes_json_serializer)
```

Comment:
This is part of a workaround to allow the use of the `api_client.deserialize()` public method instead of the private method `api_client._ApiClient__deserialize()`. See https://github.com/kubernetes-client/python/issues/977#issuecomment-592030030 for more context and LMK if we should drop the Response faking and just go with the private method instead